### ISSUE TITLE
Chisel project types / use cases

### ISSUE THREAD

**Issue by luboskmetko** [xfiveco/generator-chisel#272]
@jakub300 @arturkot @marcinkrzeminski I was thinking if Chisel could be used as a static site generator. Not sure what it takes and what are typical requirements for such generators but this could be an interesting use case and could help us to popularize the project.

So then we would have 3 project types:

- WordPress
- Static site
- Templates List

Templates list would be the current front-end option with project index.

What you think?

**Comment by jakub300** [comment]
Yesterday when I was looking at tools used by different projects i thought about this too but instead of separated project type I was thinking about that as a "feature" for front-end type but it could be separated type too.

I would recommend markdown as language for pages content and twig would be used for templating/theming.

**Comment by marcinkrzeminski** [comment]
In my opinion there is no need to separate front-end projects and static site. At the end front-end project is a static site, isn't it? From my point of view it looks like Chisel within the context of front-end project is kind of static site generator. If I'm missing something that differs front-end project from static site please let me know ;). 

Makrdown is nice indeed in terms of writing content fast but it should be optional. If someone would like to write plain HTML or use some Twig partials this should be also possible.

**Comment by luboskmetko** [comment]
Yes, the front-end is just a static site with a special home page template :) So ideally there is just a little technical modification between those two, but calling it static site generator makes it clearer from user or marketing perspective.

I'm wondering how static site generators deal with permalinks, do they take some page files and generate subfolders with index.html files in them?

**Comment by luboskmetko** [comment]
So the project types could be:

- WordPress
- Static Site
  - Create home page with list of pages

**Comment by jakub300** [comment]
I wouldn't call it small assuming that it is supposed to work well (and support markdown).
* Generating of the list must be part of twig so proper menu can be created as part of the "theme".
* Pages must support subpages (I would suggest separated directory `pages`).
* Pages should be able to specify metadata and template (similarly to WordPress)
* There need to be ability to control order of pages.

**Comment by marcinkrzeminski** [comment]
I guess it's up to us to decide what should be part of the static site generator but @jakub300 is right. There is couple of things that aren't small changes, but this go on a roadmap and be implemented in multiple releases.

As of site structure / permalinks for example in Jekyll you could configure it I guess. Probably the same goes for other generators. We should do a research on the most popular (seems like @jakub300 already did a little bit of research) ones and see what would work best for us.

**Comment by luboskmetko** [comment]
I meant that if we implement static site generator, then static site with special home page with list of all pages should be just a small modification from it, shouldn't it? Of course implementing static site generator itself is much bigger task.

**Comment by arturkot** [comment]
I'd add multi language support to what @jakub300 has written above. Not that uncommon when you design a blog or a home page. Creating a static website generator is quite a challenge IMHO! Not to mention there's a few quite popular solutions already: https://www.staticgen.com/

Hm. And what about making Chisel "static website generator" compatible? Chisel could handle JS stuff, assets, possibly SCSS (not sure how fast it's in Jekyll for example, might still use Ruby Sass), and more complex templating with Twig. Then the generated files could be intercepted by a static website generator which would do its job.

I mean that Chisel is good in one thing – project scaffolding. Static website generators, on the other hand, have other responsibilities. For example it's not trivial to setup ES6 in Jekyll. We could make that easier!

**Comment by luboskmetko** [comment]
Ok, so what if we start with a less ambitious goal - one common use-case is to create a simple static site or maybe SPA / PWA. For beginning it would be nice if we can run such static site directly at `localhost:3000` instead of having Chisel project index there. What it would take to make this work while still having an option to use standard project index?

**Comment by luboskmetko** [comment]
Just play with it and to make it work like static site served from `dist`, we just need to update Browsersync config

```
    var browserSyncConfig = {
      server: './dist',
      ghostMode: false,
      online: true
    }
```

and create Index page with `yo chisel:page "Index"`

I think it would be worth to have this option as it shouldn't be much work. We could then have these options

- WordPress
- Static Site
- Static Site with Project Index

If they choose Static site, we could display a checkbox PWA and generate manifest file.

**Comment by luboskmetko** [comment]
For me personally having just the above covered in Chisel would be enough, because the truth is that if you want a real static site generator there are many advanced solutions there.
